### DISCUSSION TITLE
Deep Integration in Qdrant

### DISCUSSION THREAD

**Discussion by do-me** [qdrant/fastembed#117] [General]
Hi folks! 
At the moment I can use fastembed only in Qdrant (Python) client or as standalone service, is that right? 
Please correct me if I got the scope or the current functionality of fastembed wrong, but from my perspective, a typical setup with Qdrant needs three parts: 

1. Qdrant as DB & API 
2. Some service for embeddings, needed for initial indexing and **querying** (e.g. fastembed)
3. Some frontend (e.g. Qdrant Web UI)

It's very nice that Qdrant provides all three components, but the integration could be deeper. I think fastembed is not that crucial for a normal data processing pipeline to create the initial embeddings for the collection. However, where I would see the biggest value would be in retrieval! 

Wouldn't it be great if you could use the Qdrant API (& clients) for querying collections without the need for a query vector? Very concretely I am thinking of a new parameter for the Qdrant API where instead of the `vector` you can simply pass a `query` IF fastembed is installed. If it's not installed it should of course throw an error. 

I think such a simple integration would provide lots of value to end users and facilitate the entire setup. Also, if you add the model name in the Qdrant collection metadata, fastembed could initialize itself automatically depending on the queried collection.

Also, it would allow for very convenient improvements in Qdrant Web UI allowing for actual querying (without providing the query vectors) in the console. 

Any thought on that?

**Comment by hiepxanh** [comment]
Correct, Im unable to use it on javascript server or though api, it should have docker support with API so we can restart if crash or something

**Reply by hiepxanh** [reply]
@do-me  https://github.com/michaelfeil/infinity this one have fastembed which should satisfy our request

**Reply by do-me** [reply]
Thanks for your reply, but the problem - at least for me - is not the lack of good inference servers but rather their integration. I would like to be able to query Qdrant without having to take care of the vector before. If you find a convenient solution for this without too much overhead, let me know! 
I just published https://do-me.github.io/qdrant-frontend for this - it inferences right in your browser with a model of your choice!

**Reply by hiepxanh** [reply]
I wish it have a server interface, so we can build docker and ship it, I end up to use transformer.js too
